clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -5166,6 +5166,71 @@ test_unscheduled_initially_roi()
 #endif
 }
 
+static void
+test_unscheduled_small_timeout()
+{
+    // Test that a small timeout scaled to 0 does not turn into an infinite timeout.
+    std::cerr << "\n----------------\nTesting unscheduled input with small timeout\n";
+    static constexpr int NUM_OUTPUTS = 1;
+    static constexpr int WAIT_TIMEOUT = 5;
+    static constexpr double BLOCK_SCALE = 0.1;
```

Comment:
I think the intent here is for WAIT_TIMEOUT * BLOCK_SCALE to get converted to 0, but the constants you chose multiply to 0.5, and a _rounding_ conversion could round that up to 1.  Regardless of how the scheduler actually handles fractions, I suggest using constants that unambiguously round/truncate to 0, e.g. 4 and 0.1, or 5 and 0.01, just to eliminate that tiny bit of ambiguity.

Reply:
The scheduler has a cast from double to int which always rounds down.  But agreed may as well tweak this to avoid any confusion.